library/core/src/ptr/non_null.rs

```diff
@@ -1918,4 +1926,53 @@ mod verify {
         // Trigger panic
         let offset = nonnull_xptr.align_offset(invalid_align);
     }
+
+    #[kani::proof_for_contract(NonNull::replace)]
+    pub fn non_null_check_replace() {
+        let mut x: i32 = kani::any();
+        let mut y: i32 = kani::any();
+
+        if let nonnull_origin_ptr = NonNull::new(&mut x as *mut i32) {
```

Comment:
The `if let` here is confusing. I think the pattern is irrefutable, that is to say the if can never fail. Both because the pattern `nonnull_origin_ptr` can catch anything, and because  we know by construction that the pointer is not null.

Did you mean `if let Some(nonnull_origin_ptr) = ...`. Or maybe you meant `let nonnull_origin_ptr = NonNull::new(&mut x as *mut i32).unwrap()`